inference-engine/src/transformations/include/transformations/convert_opset1_to_legacy/convert_mul_or_add_finally.hpp

```diff
@@ -93,9 +96,43 @@ ngraph::graph_rewrite_callback get_callback() {
             }
         }
 
-        // Check that eltwise is not useless otherwise we remove it
-        if ((std::is_same<T, ngraph::opset1::Add>() && ngraph::op::util::constantIsEqualTo(const_node, 0)) ||
-            (std::is_same<T, ngraph::opset1::Multiply>() && ngraph::op::util::constantIsEqualTo(const_node, 1))) {
+        auto constant_broadcast_output = [](ngraph::PartialShape data_pshape, ngraph::Shape const_shape) -> bool {
```

Comment:
Why does this function constant specific? It may work for two dynamic tensors.

Reply:
Because if data input broadcasts constant that's ok and we can remove operation. But otherwise we can't.